### ISSUE TITLE
Postgres - parse error with json_value function

### ISSUE THREAD

**Issue by inklesspen** [sqlfluff/sqlfluff#7306]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff fails to parse this json_value function call (which is accepted very happily by postgres itself). This may be related to #7224.

```sql
CREATE TABLE audit_log_entries (
    id bigint NOT NULL,
    guild_id bigint NOT NULL,
    raw_data jsonb NOT NULL,
    action_type smallint GENERATED ALWAYS AS (json_value(raw_data, '$."action_type"' RETURNING smallint)) STORED NOT NULL,
    target_id bigint GENERATED ALWAYS AS (json_value(raw_data, '$."target_id"' RETURNING bigint)) STORED,
    logged_at timestamp(3) with time zone GENERATED ALWAYS AS (to_timestamp(((id::bit(64) >> 22)::bigint + 1420070400000) / 1000)) STORED NOT NULL
);
```

`sqlfluff lint --dialect postgres sqlfluff_test.sql`

### Expected Behaviour

I expected the sql to be parsed correctly.

### Observed Behaviour

```
$ sqlfluff lint --dialect postgres sqlfluff_test.sql 
== [sqlfluff_test.sql] FAIL                                                     
L:   5 | P:   5 | LT05 | Line is too long (122 > 80).
                       | [layout.long_lines]
L:   5 | P:  57 |  PRS | Line 5, Position 57: Found unparsable section:
                       | '(raw_data, \'$."action_type"\' RETURNING s...'
L:   6 | P:   5 | LT05 | Line is too long (105 > 80).
                       | [layout.long_lines]
L:   6 | P:  53 |  PRS | Line 6, Position 53: Found unparsable section:
                       | '(raw_data, \'$."target_id"\' RETURNING big...'
L:   7 | P:   5 | LT05 | Line is too long (146 > 80).
                       | [layout.long_lines]
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

Run `sqlfluff lint --dialect postgres` with the SQL file given above.

### Dialect

Postgres

### Version

sqlfluff, version 3.5.0
Python 3.12.11

### Configuration

No config files; ran command as `sqlfluff lint --dialect postgres sqlfluff_test.sql`

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
